src/components/Button.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+import * as React from 'react';
+import styled from 'styled-components';
+
+type ClickHandler = (el: HTMLElement, ev: MouseEvent) => void;
+type FocusHandler = (el: HTMLElement, ev: MouseEvent) => void;
+type BlurHandler = (el: HTMLElement, ev: MouseEvent) => void;
+type KeyDownHandler = (el: HTMLElement, ev: KeyboardEvent) => void;
+
+type Props = {
+  className: string,
+  ariaLabel: ?string,
+  onClick: ClickHandler,
+  onFocus?: FocusHandler,
+  onBlur?: BlurHandler,
+  onKeyDown?: KeyDownHandler,
+  children?: React$Node,
+};
+
+class Button extends React.Component<Props> {
+  button: ?HTMLButtonElement;
+
+  focus() {
+    if (!this.button) {
+      return;
+    }
+
+    this.button.focus();
+  }
+
+  render() {
+    return (
+      <button
+        ref={button => (this.button = button)}
+        className={this.props.className || ''}
+        onBlur={this.props.onBlur}
+        onFocus={this.props.onFocus}
+        onClick={this.props.onClick}
+        onKeyDown={this.props.onKeyDown}
+        aria-label={this.props.ariaLabel}
```

Comment:
Shouldn`t this be named `ariaLabel`? And why not simply passing down all props (including children) with `<button {...props} />`?

Reply:
I just copied CloseLink, I have to admit :)

Reply:
`aria-label` is aligned with how react treats these - see https://reactjs.org/docs/accessibility.html

Reply:
Again, you are right. Just strange they do not conform to the rest of the props they use (`className`, `htmlFor` etc.). Anyway, passing down all props could make this component more reusable I guess. But we can leave it as it is for now if you think this is not part of this PR.

Reply:
-1 for not passing down all props.

Reply:
It seems passing all props to an html component is considered an anti pattern - https://vasanthk.gitbooks.io/react-bits/anti-patterns/07.spreading-props-dom.html

Also I cannot safely type this with flow, typescript at least has prop types :/